osbuild/meta.py

```diff
@@ -410,6 +458,47 @@ def _parse_caps(cls, _klass, _name, node):
 
     @classmethod
     def load(cls, root, klass, name) -> Optional["ModuleInfo"]:
+        base = cls.MODULES.get(klass)
+        if not base:
+            raise ValueError(f"Unsupported type: {klass}")
+        path = os.path.join(root, base, name)
+
+        try:
+            return cls._load_from_json(path, klass, name)
+        except FileNotFoundError:
+            pass
+        return cls._load_from_py(path, klass, name)
+
+    @classmethod
+    def _load_from_json(cls, path, klass, name) -> Optional["ModuleInfo"]:
+        # ideas welcome for a better filename/suffix :)
+        meta_json_suffix = ".meta-json"
+        with open(path + meta_json_suffix, encoding="utf-8") as fp:
+            meta = json.load(fp)
+
+        schema = Schema(META_JSON_SCHEMA, "meta.json validator")
+        res = schema.validate(meta)
+        if not res.valid:
+            # XXX: should we raise an exception instead?
```

Comment:
Yes, if we are parsing a schema JSON and it is incorrect we should communicate to the user why the module is being ignored. I can be convinced that it shouldn't be an exception but there should at least be some output indicating that this is the case.

Note that `load_from_py` also excepts in that case I believe.

Reply:
Personally, I would keep the behavior consistent with the case when the schema is part of the Python implementation.

Reply:
I can write a test to double check but AFAIK the osbuild code at this _load_from_py will not fail with an exacption for an invalid schema, just for invalid json. I can align the two so that there is a syntax error if the json parsing fails but not if schema validation fails (and ideally a test that verifies that both behave exactly the same) (?)